Sources/SmokeDynamoDB/AWSDynamoDBCompositePrimaryKeyTable+updateItems.swift

```diff
@@ -306,29 +306,33 @@ public extension AWSDynamoDBCompositePrimaryKeyTable {
         do {
             try await self.writeTransactionItems(entries, constraints: constraints)
         } catch DynamoDBError.transactionCanceled(let exception) {
-            let reasons = try exception.cancellationReasons?.compactMap { cancellationReason -> SmokeDynamoDBError? in
+            guard let cancellationReasons = exception.cancellationReasons else {
+                throw SmokeDynamoDBError.transactionCanceled(reasons: [])
+            }
+            
+            let keys = entries.map { $0.compositePrimaryKey } + constraints.map { $0.compositePrimaryKey }
+            
+            let reasons = try zip(cancellationReasons, keys).compactMap { (cancellationReason, entryKey) -> SmokeDynamoDBError? in
                 let key: StandardCompositePrimaryKey?
                 if let item = cancellationReason.item {
                     key = try DynamoDBDecoder().decode(.init(M: item))
                 } else {
                     key = nil
                 }
                 
+                let partitionKey = key?.partitionKey ?? entryKey?.partitionKey
+                let sortKey = key?.sortKey ?? entryKey?.sortKey
+                
                 // https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_ExecuteTransaction.html
                 switch cancellationReason.code {
                 case "None":
                     return nil
                 case "ConditionalCheckFailed":
-                    if let key = key {
-                        return SmokeDynamoDBError.conditionalCheckFailed(partitionKey: key.partitionKey,
-                                                                         sortKey: key.sortKey,
-                                                                         message: cancellationReason.message)
-                    } else {
-                        return SmokeDynamoDBError.transactionUnknown(code: cancellationReason.code, partitionKey: nil,
-                                                                     sortKey: nil, message: cancellationReason.message)
-                    }
+                    return SmokeDynamoDBError.conditionalCheckFailed(partitionKey: partitionKey ?? "Not provided",
+                                                                     sortKey: sortKey ?? "Not provided",
```

Comment:
Is this preferable to having an optional property? What if some DDB has primary key that's `Not provided`? :)